### DISCUSSION TITLE
Time for a new release? (v12)

### DISCUSSION THREAD

**Discussion by martinwhitaker** [steveicarus/iverilog#828] [General]
The last release was over 2 years ago, so it would be good to have a new one. Many fixes haven't been back-ported to the v11 branch, so I would suggest going straight to v12.

**Comment by caryr** [comment]
I'm not against this, but if we get approved for the SDF work under GSOC then we will also want to release that later this year. As a minimum we should target a new major release in the coming year. Preferably early fall.

I think the SDF work will require a number of fixes/tweaks internal to the compiler to function correctly. So while visibly it will mostly focus on missing implementations of timing related constructs, it will also require fixes for some more fundamental issues/bugs. I hope to have a concrete proposal for review over the next couple weeks.

**Comment by steveicarus** [comment]
We might actually want to make a stable release *before* the GSoC SDF work.
Get something stable out there before ripping it all up.Something to think
about.

We can always make a 12.1 if the SDF work goes smoothly.


On Fri, Dec 23, 2022 at 12:50 PM Cary R. ***@***.***> wrote:

> I'm not against this, but if we get approved for the SDF work under GSOC
> then we will also want to release that later this year. As a minimum we
> should target a new major release in the coming year. Preferably early fall.
>
> I think the SDF work will require a number of fixes/tweaks internal to the
> compiler to function correctly. So while visibly it will mostly focus on
> missing implementations of timing related constructs, it will also require
> fixes for some more fundamental issues/bugs. I hope to have a concrete
> proposal for review over the next couple weeks.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/steveicarus/iverilog/discussions/828#discussioncomment-4487798>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAACKMW4X4FTS4XMIDKUYGTWOXQ4XANCNFSM6AAAAAATHBJBI4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Steve Williams
***@***.*** ***@***.***>

**Comment by caryr** [comment]
I'm not against that, but I think the SDF changes may be significant enough to also trigger a full release (13.0). I'm not against two major releases in a year, but that would certainly be out of character for us ;).

If we want to release the SDF at a later date then I think we should make a new release sooner rather than later. @larsclausen what is your opinion? You have been making a majority of the changes recently. Do you see your work being at a good place for a release soon?

**Reply by larsclausen** [reply]
The main thing I've been working on was type parameter support. Took almost a year of refactoring, but that's now merged. I wouldn't wait for specific changes, there are always more new features and fixes behind the horizon, just cut a release once or twice a year on a regular schedule.

**Comment by steveicarus** [comment]
Yeah, I'm leaning towards just making a new release. I can create a
release branch next week. I'll manage the process.

On Sat, Dec 24, 2022 at 12:08 PM Lars-Peter Clausen
***@***.***> wrote:
>
> The main thing I've been working on was type parameter support. Took almost a year of refactoring, but that's now merged. I wouldn't wait for specific changes, there are always more new features and fixes behind the horizon, just cut a release once or twice a year on a regular schedule.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.Message ID: ***@***.***>



-- 
Steve Williams
***@***.***

**Comment by steveicarus** [comment]
I have created the v12-branch branch, and moved the master branch to v13 devel. I think I got the infrastructure set up properly, at least for builds and actions. Need to remind myself how to "deploy" the new version, but at least the branches and tags are there.

BTW I fleshed out the iverilog/wiki page to summarize whats going on.

**Comment by steveicarus** [comment]
OK, this took a bit longer than I expected, and I made more mistakes along the way than I should have, but I think I got it now. I've created the v12-branch and got it down the path of being the stable release, and I've converted master into being development for the next (v13) release. I also improved the documentation around releases, including creating a github wiki page to explain the git infrastructure for release, and rest documentation for release related files in the source.

And I've created a v12_0 tag to mark the starting point for the v12 release.

**Reply by martinwhitaker** [reply]
Various places still to update:
  - iverilog.icarus.com (points to ftp.icarus.com, which isn't currently accessible, and also to http://iverilog.wikia.com/wiki/Installation_Guide)
  - https://sourceforge.net/projects/iverilog/files/ (shows 79 downloads this week, so people are still going there)

I guess we should work on moving the rest of the old wiki content here and closing that down.

**Reply by caryr** [reply]
I have copied the v12 tar.gz file to sourceforge and created a release here for the v12_0 tag. A little late, but at least it's now more obvious we made a new release in December.

**Reply by caryr** [reply]
I had to update the file on sourceforge a bit since that normally has had autoconf.sh run which is not the case for the github files.

**Comment by caryr** [comment]
Closing this since this has already been done.
